packages/nimble-components/src/table/models/keyboard-navigation-manager.ts

```diff
@@ -0,0 +1,1025 @@
+                    handled = this.onUpArrowPressed();
+                    break;
+                case keyArrowDown:
+                    handled = this.onDownArrowPressed();
+                    break;
+                default:
+                    break;
+            }
+        }
+        if (handled) {
+            event.preventDefault();
+        }
+
+        // TODO: unhandled keys: PageUp/PageDown, Home/End, Ctrl-Home/End
+        // Enter/F2/Ctrl-Enter (nav mode => cell or action menu interaction modes)
+    };
+
+    private readonly onKeyDown = (event: KeyboardEvent): void => {
+        if (
+            !this.inNavigationMode
+            && event.key === keyEscape
+            && !event.defaultPrevented
+            && (this.focusState.focusType === TableFocusType.cellActionMenu
+                || this.focusState.focusType === TableFocusType.cellContent)
+        ) {
+            const focusedRow = this.getCurrentRow();
+            if (focusedRow) {
+                this.focusState.focusType = TableFocusType.cell;
+                this.focusRowElement(focusedRow);
+                this.inNavigationMode = true;
+                event.preventDefault();
+            }
+        }
+        // TODO: unhandled keys: Enter to exit cell interaction mode (according to HLD at least - may not make sense until have text inputs/ combobox in a cell)
+    };
+
+    private onLeftArrowPressed(): boolean {
+        const focusType = this.focusState.focusType;
+        let focusedRow: TableRow | TableGroupRow | undefined;
+        let focusedRowElements: TableRowFocusableElements;
+        if (this.hasRowOrCellFocusType()) {
+            focusedRow = this.getCurrentRow();
+            focusedRowElements = focusedRow!.getFocusableElements();
+        }
+
+        switch (focusType) {
+            case TableFocusType.headerActions:
+                if (this.focusState.headerActionIndex! - 1 < 0) {
+                    return false;
+                }
+                this.focusState.headerActionIndex! -= 1;
+                this.focusHeaderElement();
+                return true;
+            case TableFocusType.columnHeader:
+                if (this.focusState.columnIndex! - 1 >= 0) {
+                    this.focusState.columnIndex! -= 1;
+                    this.focusHeaderElement();
+                    return true;
+                }
+                if (this.focusableHeaderElements.headerActions.length > 0) {
+                    this.focusState.focusType = TableFocusType.headerActions;
+                    this.focusState.headerActionIndex = this.focusableHeaderElements.headerActions.length - 1;
+                    this.focusHeaderElement();
+                    return true;
+                }
+                return false;
+            case TableFocusType.row:
+                if (this.isRowExpanded(focusedRow) === true) {
+                    this.toggleRowExpanded(focusedRow!);
+                    this.focusRowElement(focusedRow!);
```

Comment:
Just curious, but in this case isn't the row already focused? What's the reason for setting focus again? I think this question applies for other cases as well.

Reply:
You're right, this particular call was a no-op, I'll remove it. Calling `focusRowElement()` again is only needed when we're focusing something new inside the row.

Reply:
Pushed the updates that remove that no-op call.